src/app/api/dashboard/[type]/route.js

```diff
@@ -0,0 +1,319 @@
+            });
+
+          output.push({
+            links: formattedLinks,
+            members: formattedNames,
+            discords: formattedDiscords,
+            uids: formattedUids,
+            status,
+            uid: doc.id,
+            selected: false,
+            hidden: false,
+            timestamp: timestamp || new Date(),
+          });
+        });
+        break;
+      default:
+        throw "page doesn't exist";
+    }
+
+    const sorted = output.sort((a, b) =>
+      a.timestamp.seconds < b.timestamp.seconds ? 1 : -1
+    );
+    return res.json({ message: "OK", items: sorted }, { status: 200 });
+  } catch (err) {
+    return res.json(
+      { message: `Internal Server Error: ${err}` },
+      { status: 500 }
+    );
+  }
+}
+
+export async function PUT(req, { params }) {
+  const res = NextResponse;
+  const { objects, status } = await req.json();
+  const { auth, message } = await authenticate(AUTH.PUT);
+
+  if (auth !== 200) {
+    return res.json(
+      { message: `Authentication Error: ${message}` },
+      { status: auth }
+    );
+  }
+  try {
+    const batch = writeBatch(db);
+    switch (params.type) {
+      case "admins":
+      case "committees":
+      case "judges":
+      case "mentors":
+      case "volunteers":
+      case "participants":
+      case "interests":
+      case "sponsors":
+        objects.forEach((object) => {
+          batch.update(doc(db, "users", object.uid), {
+            [`roles.${params.type}`]: status,
+          });
+          SG.send({
+            to: object.email,
+            template_id:
+              status === 1
+                ? process.env.SENDGRID_ACCEPTANCE_TEMPLATE
+                : process.env.SENDGRID_REJECTION_TEMPLATE,
+            dynamic_template_data: {
+              name: object.name,
+              position: params.type.slice(0, -1).toUpperCase(),
+            },
+          });
+        });
+        break;
+      case "feedback":
+        objects.forEach((object) => {
+          batch.update(doc(db, "feedback", object.uid), {
+            status: status,
+          });
+        });
+        break;
+      case "teams":
+        objects.forEach((object) => {
+          batch.update(doc(db, "teams", object.uid), {
+            status: status,
+          });
+        });
+        break;
+      default:
+        throw "page doesn't exist";
+    }
+    await batch.commit();
+    return res.json({ message: "OK" }, { status: 200 });
+  } catch (err) {
+    return res.json(
+      { message: `Internal Server Error: ${err}` },
+      { status: 500 }
+    );
+  }
+}
+
+export async function DELETE(req, { params }) {
+  const res = NextResponse;
+  const { auth, message } = await authenticate(AUTH.DELETE);
+  const objects = req.nextUrl.searchParams.get("remove").split(",");
+
+  if (auth !== 200) {
+    return res.json(
+      { message: `Authentication Error: ${message}` },
+      { status: auth }
+    );
+  }
+  try {
+    const batch = writeBatch(db);
```

Comment:
dont do batch, parallel writes r faster which is what we will be changing over to